### PR TITLE
[unit-test-cmake] Fix function name and feature name

### PR DESCRIPTION
**Pull request by JackBoosY**

### CODE REVIEW

**Comment by dg0yt** [comment]
`host-path-list` is not in the default features.
And there is couple of errors with it now (x64-linux):
~~~
-- vcpkg_host_path_list(PREPEND var d e) resulted in the wrong value for var;
    expected: "d;e;a;b"
    actual  : "a;b;d;e"
-- utcve_variable: var
-- utcve_variable: var
-- vcpkg_host_path_list(APPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: "a:b"
    actual  : "a:b:d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var} d e) resulted in the wrong value for ENV{var};
    expected: "d:e:a:b"
    actual  : "a:b:d:e:d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: "a:b"
    actual  : "a:b:d:e:d:e"
-- utcve_variable: var
-- utcve_variable: var
-- vcpkg_host_path_list(APPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: ""
    actual  : "d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var} d e) resulted in the wrong value for ENV{var};
    expected: "d:e"
    actual  : "d:e:d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: ""
    actual  : "d:e:d:e"
-- utcve_variable: var
-- utcve_variable: var
-- vcpkg_host_path_list(APPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: ""
    actual  : "d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var} d e) resulted in the wrong value for ENV{var};
    expected: "d:e"
    actual  : "d:e:d:e"
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var}) resulted in the wrong value for ENV{var};
    expected: ""
    actual  : "d:e:d:e"
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- vcpkg_host_path_list(PREPEND ENV{ENV{var}} d e) resulted in the wrong value for ENV{ENV{var}};
    expected: "d;e;a;b"
    actual  : "a;b;d;e"
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- utcve_variable: ENV{var}
-- vcpkg_host_path_list(APPEND ENV{ENV{var}}) failed to set ENV{ENV{var}};
    expected: ""
-- utcve_variable: ENV{var}
-- vcpkg_host_path_list(PREPEND ENV{ENV{var}}) failed to set ENV{ENV{var}};
    expected: ""
-- utcve_variable: var
-- utcve_variable: var
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var} d e) resulted in the wrong value for ENV{var};
    expected: "d:e:a:b"
    actual  : "a:b:d:e"
-- utcve_variable: var
-- utcve_variable: var
-- utcve_variable: var
-- utcve_variable: var
-- utcve_variable: var
-- utcve_variable: var
-- vcpkg_host_path_list(APPEND ENV{var}) failed to set ENV{var};
    expected: ""
-- utcve_variable: var
-- vcpkg_host_path_list(PREPEND ENV{var}) failed to set ENV{var};
    expected: ""
~~~
At least no host path separator issue...

**Comment by JackBoosY** [comment]
Oh no...

**Comment by JackBoosY** [comment]
@dg0yt `z_vcpkg_cmake_config_fixup_merge` wasn't declared in _portfile.cmake_.

**Comment by dg0yt** [comment]
> @dg0yt `z_vcpkg_cmake_config_fixup_merge` wasn't declared in _portfile.cmake_ in PR #22546.

Why should it be in `portfile.cmake`? 
It is in `vcpkg_cmake_config_fixup.cmake`, next to the maintainer function from this script ports. That's why this feature has a proper host dependency on `vcpkg-cmake-config`.

> [Remove merge-libs since it's not implement](https://github.com/microsoft/vcpkg/pull/24799/commits/c8638a49205d5e274386f2edb2c353df7339e6fc)

Revert that. It is implemented and it works. I created, and used it, to implement a change to `vcpkg-cmake-config`. Test-first.
I only didn't add `merge-libs` to default features because I followed the `host-path-list` example. It would be better to add them all (unless they aren't portable).

File: scripts/test_ports/unit-test-cmake/portfile.cmake [hunk 1]
```diff
@@ -40,6 +40,7 @@ function(unit_test_match namespace value regex)
 endfunction()
 
 function(unit_test_check_variable_unset utcvu_test utcvu_variable)
+    message(STATUS "Test ${utcvu_test}")
```
Comment (dg0yt) [inline scripts/test_ports/unit-test-cmake/portfile.cmake:43]: Note that `message` is overriden in this port. You may need to use `_message`.

**Comment by JackBoosY** [comment]
In favor of https://github.com/microsoft/vcpkg/pull/24808

**Pull request closed without merging**